frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import { withRouter } from "react-router";
+import { connect } from "react-redux";
+import cx from "classnames";
+import _ from "underscore";
+import type { Location } from "history";
+
+import TitleAndDescription from "metabase/components/TitleAndDescription";
+
+import { useOnMount } from "metabase/hooks/use-on-mount";
+import { getSetting } from "metabase/selectors/settings";
+import { IFRAMED, initializeIframeResizer } from "metabase/lib/dom";
+import { parseHashOptions } from "metabase/lib/browser";
+
+import SyncedParametersList from "metabase/parameters/components/SyncedParametersList/SyncedParametersList";
+
+import type { Dashboard, Parameter, ParameterId } from "metabase-types/api";
+import type { ParameterValueOrArray } from "metabase-types/types/Parameter";
+import type { State } from "metabase-types/store";
+
+import { getValuePopulatedParameters } from "metabase-lib/parameters/utils/parameter-values";
+
+import LogoBadge from "./LogoBadge";
+import {
+  Root,
+  ContentContainer,
+  Header,
+  Body,
+  Footer,
+  ActionButtonsContainer,
+} from "./EmbedFrame.styled";
+import "./EmbedFrame.css";
+
+interface OwnProps {
+  className?: string;
+  name?: string;
+  description?: string;
+  dashboard?: Dashboard;
+  actionButtons?: JSX.Element[];
+  parameters?: Parameter[];
+  parameterValues?: Record<ParameterId, ParameterValueOrArray>;
+  setParameterValue?: (parameterId: ParameterId, value: any) => void;
+  children: React.ReactNode;
+}
+
+interface StateProps {
+  hasEmbedBranding: boolean;
+}
+
+type Props = OwnProps &
+  StateProps & {
+    location: Location;
+  };
+
+interface HashOptions {
+  bordered?: boolean;
+  titled?: boolean;
+  theme?: string;
+  hide_parameters?: string;
+  hide_download_button?: boolean;
+}
+
+function mapStateToProps(state: State) {
+  return {
+    hasEmbedBranding: !getSetting(state, "hide-embed-branding?"),
+  };
+}
+
+function EmbedFrame({
+  className,
+  children,
+  name,
+  description,
+  dashboard,
+  actionButtons,
+  location,
+  hasEmbedBranding,
+  parameters,
+  parameterValues,
+  setParameterValue,
+}: Props) {
+  const [hasInnerScroll, setInnerScroll] = useState(true);
+
+  useOnMount(() => {
+    initializeIframeResizer(() => setInnerScroll(false));
+  });
+
+  const {
+    bordered = IFRAMED,
+    titled = true,
+    theme,
+    hide_parameters,
+    hide_download_button,
+  } = parseHashOptions(location.hash) as HashOptions;
+
+  const showFooter =
+    hasEmbedBranding || (!hide_download_button && actionButtons);
+
+  const finalName = titled ? name : null;
+
+  const hasParameters =
+    Array.isArray(parameters) &&
+    parameters.length > 0 &&
+    typeof setParameterValue === "function";
```

Comment:
nit: I think checking if the variable exists should be enough. checking its type makes it looks like it could be another type. But after I check its TS type, it can only be a function. What do you think?

```suggestion
    setParameterValue;
```